### DISCUSSION TITLE
Sentences not properly separated

### DISCUSSION THREAD

**Discussion by sebnun** [deepgram/community#193] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

Hi, 

I'm trying to get sentences from the DG Whisper API (to replicate Openai's API segments response), but it looks like some sentences are sent together in the response as a single sentence.

Example sentences from DG Whisper

> Dagen är kommen, jag och Malin ska hoppa fallskärm Vi är 8 pers, 8 svenskar till och med som ska hoppa fallskärm idag Här I Wollongong Och jag har typ pratat om det här sedan dag ett I Australien att jag vill ha på fallskärm Och tyst du med?
####################################
Eller är det jag som har lurat med dig på det här?
####################################
Jag tror att du har lurat med mig mest Men det bara känns som en sån här grej man måste göra när man är I Australien Jag vet inte varför Och jag som är extremt höjdröt också Men det blir kul Vi Ska möta upp de andra Och hoppa Hjälp säger jag bara Hjälp Sen har vi en liten rackare som håller på med instastory Det är så jävla konstigt Molly Jag vet Jag har typ dålig magen Jag har tvungna sätta mig ner för att jag skakar typ I hela kroppen Det här blir kul Jag har så jäkla mycket handsvett Blandat med dålig magen Blandat med skitsvett I allmänhet Malin vad har vi gjort Vad har vi gjort Så efterblivande Det är så jävla efterblivande Alltså vi har typ betalat 3000 spänn för det här Det ser inte så bra ut Är vi taggade gänget?
####################################
Jaaa!
####################################
Hur känns det?

Example sentences from Openai

> Dagen är kommen, jag och Malin ska hoppa fallskärm.
####################################
 Vi är 8 pers, 8 svenskar till och med som ska hoppa fallskärm idag.
####################################
 Här i Wollongong och jag har inte pratat om det här sedan dag ett i Australien att jag vill hoppa fallskärm.
####################################
 Och tydligtvis du med. Eller är det jag som har lurat med dig på det här?
####################################
 Jag tror att du har lurat med mig mest.
####################################
 Men det bara känns som en grej man måste göra när man är i Australien, jag vet inte varför.
####################################
 Nej faktiskt.
####################################
 Och jag som är extremt höjdröt också men det blir kul.
####################################
 Det blir kul.
####################################
 Ska möta upp de andra och hoppa.
####################################
 Hjälp säger jag bara. Hjälp.
####################################
 Sen har vi en liten rackare som håller på med instastory.
####################################
 Det är så jävla konstigt Malin.
####################################
 Jag vet.
####################################
 Jag har typ dålig magen.
####################################
 Jag har tvungen att sätta mig ner för att jag skakar typ i hela kroppen.
####################################
 Det här blir kul.
####################################
 Jag har så jäkla mycket handsvett blandat med dålig magen.
####################################
 Blandat med skitsvett i allmänhet.
####################################
 Malin vad har vi gjort?
####################################
 Vad har vi gjort?
####################################
 Så efterblivande.
####################################
 Det är så jävla efterblivande.
####################################
 Alltså vi har typ betalat 3000 spänn för det här.
####################################
 Det är så jävla jävla kul.
####################################
 Åh!
####################################
 Är vi taggade gänget?
####################################
 Ja!
####################################
 Hur känns det?

### If possible, please attach your code or paste it into the text box.

```
deepgram.transcription
    .preRecorded(source, {
      punctuate: true,
      paragraphs: true,
      model: "whisper-large",
      detect_language: true,
    })
    .then((response) => {
      let transcription = "";

      response.results.channels[0].alternatives[0].paragraphs.paragraphs.forEach(
        (p) => {
          p.sentences.forEach((s) => {
            transcription += s.text;
            transcription += "\n####################################\n";
          });
        }
      );

      fs.writeFileSync("deepgram.txt", transcription);
    })
```

### If possible, please attach an example audio file to reproduce the issue.

Audio to reproduce: https://pub-a97f2be9f4f04cabb59e384d045bdcdf.r2.dev/audio.webm

**Comment by jjmaldonis** [comment, marked as answer]
Hi @sebnun, we are currently working on improving functionality such as this to provide better punctuation, paragraphs, and overall formatting. Some of our other models perform better, and improvement are happening quickly.

It looks like you're transcribing Swedish, which is supported by the `base` and `enhanced` tiers in Deepgram. Do you get better formatting results when using those tiers rather than whisper?

**Comment by sebnun** [comment]
I tried the nova model with the tier base and unfortunately the word error rate increases and I still see some issues with the sentences.
I don't have access and cannot justify the cost of the enhanced tier for my use case.

> Dagen är kommen, jag och Malin, ska vara på film Vi är åtta pers åtta svenskar till och med som ska hoppa fallskärm idag här I Bornholm.
####################################
Och jag har inte pratat om det som dag ett I Stralen, att jag vill åka på Falkenberg.
####################################

A suggestion: I really like the speed and price of the DG Whisper API, it would be nice if we had the option to get the response straight out of whisper (i.e. with "segments")
